python/src/sdf/pyModel.cc

```diff
@@ -0,0 +1,212 @@
+         "Get the number of explicit model that are immediate (not nested) "
+         "children of this Model object.")
+    .def("model_by_index",
+         pybind11::overload_cast<uint64_t>(
+           &sdf::Model::ModelByIndex, pybind11::const_),
+         pybind11::return_value_policy::reference,
+         "Get an immediate (not nested) child joint model on an index.")
+    .def("model_by_name",
+         pybind11::overload_cast<const std::string &>(
+           &sdf::Model::ModelByName, pybind11::const_),
+         pybind11::return_value_policy::reference,
+         "Get an immediate (not nested) mutable child model based on an "
+         "index.")
+    .def("model_name_exists", &sdf::Model::ModelNameExists,
+         "Get whether a model name exists.")
+    .def("raw_pose", &sdf::Model::RawPose,
+         "Get the pose of the model. This is the pose of the model "
+         "as specified in SDF (<model> <pose> ... </pose></model>), and is "
+         "typically used to express the position and rotation of a model in a "
+         "global coordinate frame.")
+    .def("set_raw_pose", &sdf::Model::SetRawPose,
+         "Set the pose of the model.")
+    .def("canonical_link", &sdf::Model::CanonicalLink,
+         pybind11::return_value_policy::reference,
+         "Get the model's canonical link")
+    .def("canonical_link_name", &sdf::Model::CanonicalLinkName,
+         "Get the name of the model's canonical link. An empty value "
+         "indicates that the first link in the model or the first link found "
+         "in a depth first search of nested models is the canonical link.")
+    .def("set_canonical_link_name", &sdf::Model::SetCanonicalLinkName,
+         "Set the name of the model's canonical link. An empty value "
+         "indicates that the first link in the model or the first link found "
+         "in a depth first search of nested models is the canonical link.")
+    .def("pose_relative_to", &sdf::Model::PoseRelativeTo,
+         "Get the name of the coordinate frame relative to which this "
+         "frame's pose is expressed. An empty value indicates that the frame "
+         "is expressed relative to the attached-to link.")
+    .def("set_pose_relative_to", &sdf::Model::SetPoseRelativeTo,
+         "Set the name of the coordinate frame relative to which this "
+         "frame's pose is expressed. An empty value indicates that the frame "
+         "is expressed relative to the attached-to link.")
+    .def("semantic_pose", &sdf::Model::SemanticPose,
+         "Get SemanticPose object of this object to aid in resolving "
+         "poses.")
+    .def("placement_frame_name", &sdf::Model::PlacementFrameName,
+         "Get the name of the placement frame of the model.")
+    .def("set_placement_frame_name", &sdf::Model::SetPlacementFrameName,
+         "Set the name of the placement frame of the model.")
+    .def("canonical_link_and_relative_name", &sdf::Model::CanonicalLinkAndRelativeName,
```

Comment:
Do we need special handling here with the return_value_policy since we have 
`const Link*` in the pair?